src/components/HomePage/ListOfHouses/ListOfHouses.tsx

```diff
@@ -68,26 +77,50 @@ function ListOfHouses({ houses }: { houses: BasicHouseData[] }) {
         />
       </Box>
       <Box component="div" className={styles.housesList}>
-        {sortedHouses.map((item: BasicHouseData, i) => (
-          <Box component="div" className={styles.houseElement} key={i.toString()}>
-            <h4>
-              {item.city}, {item.streetName} {item.streetNumber}
-            </h4>
-            <p className={styles.price}>
-              {item.price}zł/m<sup>2</sup>
-            </p>
-            <img src={item.photo_0 ? item.photo_0 : noPhoto} alt="House" />
-            <p className={styles.shortInfo}>{item.descriptionField}</p>
-            <Button className={styles.moreInfo}>more info</Button>
-            <Checkbox
-              color="warning"
-              {...label}
-              icon={<FavoriteBorderIcon />}
-              checkedIcon={<FavoriteIcon />}
-              className={styles.icon}
-            />
-          </Box>
-        ))}
+        {sortedHouses.map((item: BasicHouseData, i) => {
+          if (item.images.length === 0) {
+            item.images.push(noPhoto);
+          }
+          if (item.images.length === 1) {
+            slideProperties.arrows = false;
+            slideProperties.canSwipe = false;
+          }
+          return (
+            <Box component="div" className={styles.houseElement} key={i.toString()}>
```

Comment:
Can't 100% confirm or deny but casting an index to a string might still be something that React doesn't handle well. I believe the safest option here would be using the URL of an image itself as a key as this guarantees uniqueness even if it might seem a little clunky.